tests/prove_and_verify.rs

```diff
@@ -1,20 +1,57 @@
-use plonky::{verify_proof_circuit, CircuitBuilder, Curve, Field, PartialWitness, Tweedledee, Tweedledum};
+use plonky::{blake_hash_base_field_to_curve, msm_parallel, rescue_hash_1_to_1, rescue_hash_n_to_1, rescue_sponge, verify_proof_circuit, AffinePoint, Circuit, CircuitBuilder, Curve, CurveMulOp, Field, HaloCurve, PartialWitness, Tweedledee, Tweedledum, Witness};
 use std::time::Instant;
 
 // Make sure it's the same as in `plonk.rs`.
 const NUM_WIRES: usize = 9;
 
-#[test]
-fn test_proof_trivial_circuit() {
-    let mut builder = CircuitBuilder::<Tweedledee>::new(128);
-    let t = builder.constant_wire(<Tweedledee as Curve>::ScalarField::ZERO);
+fn get_trivial_circuit<C: HaloCurve>(x: C::ScalarField) -> (Circuit<C>, Witness<C::ScalarField>) {
+    let mut builder = CircuitBuilder::<C>::new(128);
+    let t = builder.constant_wire(x);
     builder.assert_zero(t);
     let mut partial_witness = PartialWitness::new();
-    partial_witness.set_target(t, <Tweedledee as Curve>::ScalarField::ZERO);
+    partial_witness.set_target(t, x);
     let circuit = builder.build();
     let witness = circuit.generate_witness(partial_witness);
-    let proof = circuit.generate_proof::<Tweedledum>(witness, true).unwrap();
-    assert!(verify_proof_circuit::<Tweedledee, Tweedledum>(&[], &proof, &circuit,).is_ok());
+    (circuit, witness)
+}
+
+#[test]
+fn test_proof_trivial_circuit() {
+    let (circuit, witness) = get_trivial_circuit(<Tweedledee as Curve>::ScalarField::ZERO);
+    let proof = circuit
+        .generate_proof::<Tweedledum>(witness.clone(), &[], true)
+        .unwrap();
+    assert!(
+        verify_proof_circuit::<Tweedledee, Tweedledum>(&[], &proof, &[], &circuit, true).is_ok()
```

Comment:
Maybe nice to use either `expect(msg)`, `unwrap()`, or `?` with a `Result` return type (as you mentioned) just so we'll see the Result's error easily upon failure.